lab-dan/model/storage.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const objectConstructor = require('./objectConstructor')
+
+let storage = {}
+storage.data = {
+  '1234-test-obj': {
+    id: '1234-test-obj',
+    creationDate: new Date(12/11/2016),
+    name: 'Test',
+    breed: 'Shiba',
+  },
+}
+
+// let fs = blueBird.promisifyAll(require('fs'), { suffix: 'Promised'})
+
+// This is what a monkeyfied fs function looks like
+// fs.readFilePromised('file').then()
+// fs.writeFilePromised('file').then()
+
+storage.fetchItem = function(id) {
+  // return fs.readFilePromised(`${__dirname}/../data/data.json`)
+  //   .then(data => {
+  //     data = JSON.parse(data.toString())
+  //     return data[id]
+  //   })
+  if (!id) {
+    return storage.fetchAll()
```

Comment:
I think I'm on the fence about using fetchAll if no ID is passed through the request for a single item. It's a design decision, but given that you have a separate route set up for getting all records, I think I would rather see an error response here.